### PR TITLE
NX-OS ACLs module

### PR DESCRIPTION
**Pull request by adharshsrivatsr**
##### SUMMARY
New resource module for nxos_acls

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
nxos_acls

### CODE REVIEW

**Review by NilashishC — CHANGES_REQUESTED** [review]
`line` key seems to be absent from the argspec.

**Comment by NilashishC** [comment]
recheck

File: lib/ansible/modules/network/nxos/nxos_acls.py [hunk 1]
```diff
@@ -0,0 +1,784 @@
...
+    state: deleted
+
+# After state:
+# ------------
+#
+# ip access-list ACl1v6
+# ip access-list ACL2v6
+#  10 deny ipv6 any 2001:db8:3000::/36
+#  20 permit tcp 2001:db8:2000:2::2/128 2001:db8:2000:ab::2/128
+
```
Comment (GomathiselviS) [inline lib/ansible/modules/network/nxos/nxos_acls.py:646]: Can you give examples of various levels of delete operation?

**Comment by mikewiebe** [comment]
@adharshsrivatsr and @NilashishC Is this PR ready for deeper review and testing?  I just tried to run the integration tests against one of our N9K testbeds but see errors.

**Comment by adharshsrivatsr** [comment]
> @adharshsrivatsr and @NilashishC Is this PR ready for deeper review and testing? I just tried to run the integration tests against one of our N9K testbeds but see errors.
> 
> Here is the error I am seeing:
> 
> ```
> 2020-02-25T14:39:06: %LIB-INFO: TASK [nxos_acls : Merged] ******************************************************************************************************************************************************************************
> 2020-02-25T14:39:06: %LIB-INFO: task path: /root/agents-ci/ansible/test/integration/targets/nxos_acls/tests/cli/merged.yml:8
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> attempting to start connection
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> using connection plugin network_cli
> 2020-02-25T14:39:06: %LIB-INFO: Found ansible-connection at path /root/agents-ci/ansible/bin/ansible-connection
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> found existing local domain socket, using it!
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> updating play_context for connection
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com>
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> local domain socket path is /root/.ansible/pc/835c2f063b
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> ESTABLISH LOCAL CONNECTION FOR USER: root
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575 `" && echo ansible-tmp-1582659540.24-127387868099575="` echo /root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575 `" ) && sleep 0'
> 2020-02-25T14:39:06: %LIB-INFO: Using module file /root/agents-ci/ansible/lib/ansible/modules/network/nxos/nxos_acls.py
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> PUT /root/.ansible/tmp/ansible-local-13278F0qMCj/tmp0GRlAk TO /root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/ /root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py && sleep 0'
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py && sleep 0'
> 2020-02-25T14:39:06: %LIB-INFO: <n9k.domain.com> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/ > /dev/null 2>&1 && sleep 0'
> 2020-02-25T14:39:06: %LIB-INFO: The full traceback is:
> 2020-02-25T14:39:06: %LIB-INFO: Traceback (most recent call last):
> 2020-02-25T14:39:06: %LIB-INFO:   File "/root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py", line 102, in <module>
> 2020-02-25T14:39:06: %LIB-INFO:     _ansiballz_main()
> 2020-02-25T14:39:06: %LIB-INFO:   File "/root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py", line 94, in _ansiballz_main
> 2020-02-25T14:39:06: %LIB-INFO:     invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
> 2020-02-25T14:39:06: %LIB-INFO:   File "/root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py", line 40, in invoke_module
> 2020-02-25T14:39:06: %LIB-INFO:     runpy.run_module(mod_name='ansible.modules.network.nxos.nxos_acls', init_globals=None, run_name='__main__', alter_sys=True)
> 2020-02-25T14:39:06: %LIB-INFO:   File "/usr/lib/python2.7/runpy.py", line 188, in run_module
> 2020-02-25T14:39:06: %LIB-INFO:     fname, loader, pkg_name)
> 2020-02-25T14:39:06: %LIB-INFO:   File "/usr/lib/python2.7/runpy.py", line 82, in _run_module_code
> 2020-02-25T14:39:06: %LIB-INFO:     mod_name, mod_fname, mod_loader, pkg_name)
> 2020-02-25T14:39:06: %LIB-INFO:   File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
> 2020-02-25T14:39:06: %LIB-INFO:     exec code in run_globals
> 2020-02-25T14:39:06: %LIB-INFO:   File "/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/modules/network/nxos/nxos_acls.py", line 784, in <module>
> 2020-02-25T14:39:06: %LIB-INFO:   File "/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/modules/network/nxos/nxos_acls.py", line 779, in main
> 2020-02-25T14:39:06: %LIB-INFO:   File "/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/module_utils/network/nxos/config/acls/acls.py", line 83, in execute_module
> 2020-02-25T14:39:06: %LIB-INFO:   File "/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/module_utils/network/nxos/config/acls/acls.py", line 112, in set_config
> 2020-02-25T14:39:06: %LIB-INFO:   File "/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/module_utils/network/nxos/config/acls/acls.py", line 232, in convert_values
> 2020-02-25T14:39:06: %LIB-INFO: AttributeError: 'int' object has no attribute 'isdigit'
> 2020-02-25T14:39:06: %LIB-INFO: fatal: [n9k.domain.com]: FAILED! => {
> 2020-02-25T14:39:06: %LIB-INFO:     "changed": false,
> 2020-02-25T14:39:06: %LIB-INFO:     "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-local-13278F0qMCj/ansible-tmp-1582659540.24-127387868099575/AnsiballZ_nxos_acls.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.network.nxos.nxos_acls', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib/python2.7/runpy.py\", line 188, in run_module\n    fname, loader, pkg_name)\n  File \"/usr/lib/python2.7/runpy.py\", line 82, in _run_module_code\n    mod_name, mod_fname, mod_loader, pkg_name)\n  File \"/usr/lib/python2.7/runpy.py\", line 72, in _run_code\n    exec code in run_globals\n  File \"/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/modules/network/nxos/nxos_acls.py\", line 784, in <module>\n  File \"/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/modules/network/nxos/nxos_acls.py\", line 779, in main\n  File \"/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/module_utils/network/nxos/config/acls/acls.py\", line 83, in execute_module\n  File \"/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/module_utils/network/nxos/config/acls/acls.py\", line 112, in set_config\n  File \"/tmp/ansible_nxos_acls_payload_8BJfK6/ansible_nxos_acls_payload.zip/ansible/module_utils/network/nxos/config/acls/acls.py\", line 232, in convert_values\nAttributeError: 'int' object has no attribute 'isdigit'\n",
> 2020-02-25T14:39:06: %LIB-INFO:     "module_stdout": "",
> 2020-02-25T14:39:06: %LIB-INFO:     "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
> 2020-02-25T14:39:06: %LIB-INFO:     "rc": 1
> 2020-02-25T14:39:06: %LIB-INFO: }
> 2020-02-25T14:39:06: %LIB-INFO:
> ```

This is resolved now. The module is ready for deeper review. Please let us know if there is anything.

**Comment by adharshsrivatsr** [comment]
recheck

**Comment by adharshsrivatsr** [comment]
> `line` key seems to be absent from the argspec.

I dont think this key would be needed in NX-OS. The current argspec configures all the required parameters across the versions. Please let me know if there is anything else.

**Review by NilashishC — CHANGES_REQUESTED** [review]
Please fix Shippable issues.

**Review by GomathiselviS — CHANGES_REQUESTED** [review]

**Pull request merged**